crypto-currency/bitcoin/src/main/java/org/javamoney/shelter/bitcoin/provider/CoinbaseRateProvider.java

```diff
@@ -0,0 +1,100 @@
+package org.javamoney.shelter.bitcoin.provider;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.javamoney.moneta.convert.ExchangeRateBuilder;
+import org.javamoney.moneta.spi.AbstractRateProvider;
+import org.javamoney.moneta.spi.DefaultNumberValue;
+
+import javax.money.CurrencyUnit;
+import javax.money.convert.*;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+public class CoinbaseRateProvider extends AbstractRateProvider {
+    private static final RateType RATE_TYPE = RateType.DEFERRED;
+    private static final ProviderContext CONTEXT = ProviderContextBuilder.of("CoinbaseRateProvider", RATE_TYPE)
+            .set("providerDescription", "Coinbase - Bitcoin exchange rate provider")
+            .build();
+    private static final String DEFAULT_BASE_CURRENCY = "BTC";
+
+    private final List<String> supportedCurrencies = new ArrayList<>();
+    private final Map<String, Number> rates = new ConcurrentHashMap<>();
+
+    private final Logger log = Logger.getLogger(getClass().getName());
+
+    public CoinbaseRateProvider() {
+        super(CONTEXT);
+        loadSupportedCurrencies();
+    }
+
+    @Override
+    public ExchangeRate getExchangeRate(ConversionQuery conversionQuery) {
+        CurrencyUnit baseCurrency = conversionQuery.getBaseCurrency();
+        CurrencyUnit termCurrency = conversionQuery.getCurrency();
+        ConversionContext conversionContext = ConversionContext.of(getContext().getProviderName(), RATE_TYPE);
+
+        if (!DEFAULT_BASE_CURRENCY.equals(baseCurrency.getCurrencyCode())) {
+            throw new CurrencyConversionException(baseCurrency, termCurrency, conversionContext, "Base currency not supported: " + baseCurrency);
+        }
+
+        if (!supportedCurrencies.contains(termCurrency.getCurrencyCode())) {
+            throw new CurrencyConversionException(baseCurrency, termCurrency, conversionContext, "Term currency not supported: " + termCurrency);
+        }
+
+        loadRates();
+
+        Number rate = rates.get(termCurrency.getCurrencyCode());
+        if (rate == null) {
+            throw new CurrencyConversionException(baseCurrency, termCurrency, conversionContext, "Rate not available for currency: " + termCurrency);
+        }
+        return new ExchangeRateBuilder(conversionContext)
+                .setBase(baseCurrency)
+                .setTerm(termCurrency)
+                .setFactor(DefaultNumberValue.of(rate))
+                .build();
+    }
+
+    private void loadSupportedCurrencies() {
+        try {
+            HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
We should probably check if the httpClient is an instance of `AutoCloseable` and if so `#close()` it.See [JDK-8304165](https://bugs.openjdk.org/browse/JDK-8304165)

Reply:
Thank you for the feedback! After your suggestion, I considered using a try-with-resources construct to close the connection. However, while reviewing this issue, I realized that `java.net.http.HttpClient` is not supported in JDK 8, which is the version we’re using for this project.

Rather than upgrading the JDK version, I propose we switch to Apache HttpClient, which is compatible with JDK 8 and provides similar functionality. `CloseableHttpClient` in Apache HttpClient implements `AutoCloseable`, allowing us to use the try-with-resources construct for proper resource management.

I have added a new commit with these changes.